From 28f6c40cb1aa8f2e79fbf6da013e67322322854d Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 20 Sep 2023 17:22:54 +0200
Subject: [PATCH 0561/1141] arm64: dts: st: Fix ranges and dma-ranges dts
 warning for the stm32mp251

Due to  64-bit format the check of the dtb complain on ranges and
dma-ranges formats in the mlahb node:

stm32mp251.dtsi:2558.3-10: Warning (ranges_format): /ahb:ranges:
empty "ranges" property but its #address-cells (1) differs from / (2)

stm32mp251.dtsi:2559.3-14: Warning (dma_ranges_format):
/ahb:dma-ranges: empty "dma-ranges" property but its #address-cells (1)
differs from / (2)

- Define the ranges value as done in soc@0,
- Remove the dma-ranges instead of defining it empty,

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Iadf1c68fc567edd0561dfd7b05917606e41b70f7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/329142
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 67822b26fbd3..7430e69e501a 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2694,8 +2694,8 @@ mlahb: ahb {
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
-		ranges;
-		dma-ranges;
+		ranges = <0x0 0x0 0x0 0xfffffffc>;
+		dma-ranges = <0x0 0x0 0x0 0xfffffffc>;
 
 		m33_rproc: m33@a040000 {
 			compatible = "st,stm32mp2-m33";
-- 
2.39.2

